accord-core/src/main/java/accord/primitives/Txn.java

```diff
@@ -32,13 +32,26 @@
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
-import static accord.primitives.Routables.Slice.Overlapping;
-
 public interface Txn
 {
     enum Kind
     {
-        Read, Write;
+        Read,
+        Write,
+
+        /**
+         * A pseudo-transaction whose deps represent the complete set of transactions that may execute before it,
+         * without interfering with their execution.
+         *
+         * A SyncPoint is unique in that it agrees not only an executeAt but also a precise collection of dependencies,
+         * so that is effect on invalidation of earlier transactions is durable. This is most useful for ExclusiveSyncPoint.
```

Comment:
This last half really confuses me, `SyncPoint` shouldn't invalidate?

Reply:
An `ExclusiveSyncPoint` invalidates all `TxnId` lower than it that aren't witnessed by it.

Reply:
Ah got it, mixed up that `SyncPointer` includes both.